server/src/projects/projects.service.ts

```diff
@@ -39,9 +39,12 @@ export class ProjectsService {
       `);
       return this.overwriteCodesWithLabels(records);
     } catch(e) {
-      const errorMessage = `Error finding projects by contact ID. ${e.message}`;
+      const errorMessage = `Error finding projects by contact ID. CRM: ${e.message}`;
       console.log(errorMessage);
-      throw new HttpException(errorMessage, HttpStatus.UNAUTHORIZED);
+      throw new HttpException({
+        "code": "FIND_MANY_BY_CONTACTID_PROJECT_SERVICE",
+        "message": errorMessage,
+      }, HttpStatus.UNAUTHORIZED);
```

Comment:
I'm not sure we can assume this will always be an `unauthorized` error but I can see how the promise failure in the `try` could lead us to believe that.

If we know _why_ the promise failed, then we could infer that.

_however_ — looking at this, if CRM does not find a contact record, it will silently fail because it will return a blank array of records. Then, this error handling wouldn't even be triggered.

So, I would look into that some more... the actual logic of this throws an error if the promise fails, and assumes that promise failure is always an "unauthorized", even though that might not even be the case _and_ the situation when _no projects_ are found is not handled

Reply:
You're right, UNAUTHORIZED may not be the right code here. 

I don't think the a blank array of records should be regarded as an error. In that case, the user's dashboard just shows no projects, as is already the case.

Reply:
You're right, I was thinking about something else.

I will say though that for routes in which we request a _single_ record by ID, we should use the RESTful feature in Web API: `dcp_projects/:guid`. If it's not found, you'll get a full error message. But it's not super important. Just an ideal.